contrib/world_builder/include/world_builder/point.h

```diff
@@ -0,0 +1,167 @@
+/*
+  Copyright (C) 2018 by the authors of the World Builder code.
+
+  This file is part of the World Builder.
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation, either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef _world_builder_point_h
+#define _world_builder_point_h
+
+#include <array>
+#include <cmath>
+
+#include <world_builder/coordinate_system.h>
+
+namespace WorldBuilder
+{
+  template<int dim>
+  class Point
+  {
+    public:
+      /**
+       * Constructor. Constructs a Point at (0,0) in 2d or (0,0,0) in 3d
+       * with a Cartesian coordinate system.
+       */
+      Point();
+
+      /**
+       * Constructor. Constructs a Point from a std::array<double,dim> and
+       * a coordinate system which is set by default to Cartesian.
+       */
+      Point(const std::array<double,dim> &array, CoordinateSystem coordinate_system = CoordinateSystem::cartesian);
+
+      /**
+       * Constructor. Constructs a Point from an other Point and
+       * a coordinate system which is set by default to Cartesian.
+       */
+      Point(const Point<dim> &point, CoordinateSystem coordinate_system = CoordinateSystem::cartesian);
+
+      /**
+       * Constructor. Constructs a 2d Point from two doubles and
+       * a coordinate system which is set by default to Cartesian.
+       */
+      Point(const double x, const double y, CoordinateSystem coordinate_system = CoordinateSystem::cartesian);
+
+      /**
+       * Constructor. Constructs a 3d Point from three doubles and
+       * a coordinate system which is set by default to Cartesian.
+       */
+      Point(const double x, const double y, const double z, CoordinateSystem coordinate_system = CoordinateSystem::cartesian);
+
+      /**
+       * Destructor
+       */
+      ~Point();
+
+      Point<dim> operator=(const Point<dim> &point);
```

Comment:
should return a Point<dim>&

Reply:
That doesn't seem to work, are you sure? Let's talk when you are done with your bike ;)